lib.php

```diff
@@ -94,7 +94,7 @@ function tool_abconfig_after_config() {
 
                 foreach ($conditionrecords as $conditionrecord) {
                     $iplist = $conditionrecord['ipwhitelist'];
-                    $users = json_decode($conditionrecord['users']);
+                    $users = !empty($conditionrecord['users']) ? json_decode($conditionrecord['users']) : [];
```

Comment:
xposted via matrix, an alternative to this is:
```
$users = json_decode($conditionrecord['users'] ?? '');
```
Which will also work for both places. It still looks fairly ugly either way. I'll let you decide which you prefer.

Otherwise, changes seem OK.

Reply:
ive stuck with a conditional assignment via ternary, because it feels wrong to pass an empty string in, just to get an empty array back. While it works, I think this is the more "correct" in terms of not abusing methods